yarn-project/sequencer-client/src/tx_validator/metadata_validator.ts

```diff
@@ -51,7 +51,11 @@ export class MetadataTxValidator<T extends AnyTx> implements TxValidator<T> {
     const maxBlockNumber = target.maxBlockNumber;
 
     if (maxBlockNumber.isSome && maxBlockNumber.value < this.#globalVariables.blockNumber) {
-      this.#log.warn(`Rejecting tx ${Tx.getHash(tx)} for low max block number`);
+      this.#log.warn(
+        `Rejecting tx ${Tx.getHash(tx)} for low max block number. Tx max block number: ${
+          maxBlockNumber.value
+        }, current block number: ${this.#globalVariables.blockNumber}.`,
+      );
```

Comment:
This change helped me debug reoccurrence of the Noir issue.

Reply:
We should definitely add logs for this kind of things.